test/cases/cross-distro.sh

```diff
@@ -75,17 +75,79 @@ if ! sudo curl -s --unix-socket $APISOCKET http:///localhost/api/status > /dev/n
     exit 1
 fi
 
-if ! DISTROS=$(sudo curl -s --unix-socket $APISOCKET http:///localhost/api/v1/distros/list); then
+if ! RECOGNIZED_DISTROS=$(sudo curl -s --unix-socket $APISOCKET http:///localhost/api/v1/distros/list); then
     echo "osbuild-composer server error getting distros list. 😢"
     exit 1
 fi
 
-REMAINDER=$(echo "$DISTROS" | jq -r '.distros' | grep -v -E "$PATTERN")
-if [ -n "$REMAINDER" ]; then
+# check if the list of installed repositories by the RPM matches reality
+INSTALLED_DISTROS=$(find "/usr/share/osbuild-composer/repositories" -name '*.json' | sort)
+if [ "$INSTALLED_DISTROS" == "$DISTROS_FROM_RPM" ]; then
```

Comment:
How is INSTALLED_DISTROS different from DISTROS_FROM_RPM ? You may as well grep for `/usr/share/osbuild-composer/repositories` on line 65. 

I think the comparison here should be RECOGNIZED_DISTROS (what the code says it supports) == INSTALLED_DISTROS (what the RPM ships as repo definitions) and remove the DISTROS_FROM_RPM altogether.

Reply:
> How is INSTALLED_DISTROS different from DISTROS_FROM_RPM ? You may as well grep for /usr/share/osbuild-composer/repositories on line 65.

I suppose I was overthinking it, `INSTALLED_DISTROS` are the ones actually in the system directory and `DISTROS_FROM_RPM` are what the `rpm -qil` says it installed. The idea was that if for any reason this was not the same it would be detected because the code recognizes whatever valid repo is in `/usr/share/osbuild-composer/repositories` , I'll simplify it.